server/server.js

```diff
@@ -21,10 +26,62 @@ app.get('/', function(request, response) {
 });
 
 io.on('connection', (socket) => {
-  socket.emit('message', 'hello from server');
+  //Reply to application after receiving hello message
+  socket.on('hello', (msg)=> {
+      console.log("Message received: " + msg);
+      socket.emit('reply', "hello from the server side");
+  });
   console.log("Client connected");
 });
 
 server.listen(app.get('port'), function() {
   console.log("Spots server is running on http://localhost:%s", app.get('port'));
 });
+
+//--------------------------------------------------------------------------------------------
+//Functions
+//--------------------------------------------------------------------------------------------
+
+//Registers User to database
+//Paramters: (int, string, string, string, int, boolean, boolean, string?, permitType,
+//  string?, string?, string?, string?)
+function registerUser(userId, first, last, email, phone, gotPermit, expired, expDate,
+  permittype, make, model, color, plate){
+  var userIDRef = database.ref('UserAccounts/' + userId);
+  userIDRef.set({
+    First_Name: first,
+    Last_Name: last,
+    Email: email,
+    Phone: phone,
+    Permit: gotPermit
+  });
+
+  if(gotPermit){
+    database.ref('UserAccounts/' + userId + '/Permit').set({
+      isExpired: expired,
+      ExpDate: expDate,
+      Permit_Type: permittype
+    });
+
+    database.ref('UserAccounts/' + userId + '/Vehicle').set({
+      Make: make,
+      Model: model,
+      Color: color,
+      Plate_Number: plate
+    });
+  }
+  else{
+    database.ref('UserAccounts/' + userId + '/Permit').set({
+      isExpired: null,
+      ExpDate: null,
+      Permit_Type: None
+    });
+
+    database.ref('UserAccounts/' + userId + '/Vehicle').set({
+      Make: null,
+      Model: null,
+      Color: null,
+      Plate_Number: null
+    });
```

Comment:
Can you edit the field names to match the database? 
![image](https://cloud.githubusercontent.com/assets/10568301/24516511/d0756e50-152f-11e7-99dc-784f181d6b75.png)